daft/execution/rust_physical_plan_shim.py

```diff
@@ -10,17 +11,60 @@
     PySchema,
     PyTable,
     ResourceRequest,
+    ScanTask,
     StorageConfig,
 )
 from daft.execution import execution_step, physical_plan
 from daft.expressions import Expression, ExpressionsProjection
 from daft.logical.map_partition_ops import MapPartitionOp
 from daft.logical.schema import Schema
+from daft.runners.partitioning import PartialPartitionMetadata
 from daft.table import Table
 
 PartitionT = TypeVar("PartitionT")
 
 
+def scan_with_tasks(
+    scan_tasks: list[ScanTask],
```

Comment:
NOTE: I changed this from a single `ScanTaskBatch` to avoid having to coalesce all the tasks into one fat task